x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -304,6 +316,70 @@ private List<EsQueryExec.FieldSort> buildFieldSorts(List<Order> orders) {
         }
     }
 
+    /**
+     * Looks for the case where certain stats exist right before the query and thus can be pushed down.
+     */
+    private static class PushStatsToSource extends OptimizerRule<AggregateExec> {
+
+        @SuppressWarnings("unchecked")
+        @Override
+        protected PhysicalPlan rule(AggregateExec aggregateExec) {
+            PhysicalPlan plan = aggregateExec;
+            if (aggregateExec.child() instanceof EsQueryExec queryExec) {
+                List<?>[] lists = pushableStats(aggregateExec);
+
+                // TODO: handle case where some aggs cannot be pushed down by breaking the aggs into two sources (regular + stats) + union
+                // use the stats since the attributes are larger in size (due to seen)
+                if (lists[1].size() == aggregateExec.aggregates().size()) {
+                    plan = new EsStatsQueryExec(
+                        aggregateExec.source(),
+                        queryExec.index(),
+                        queryExec.query(),
+                        queryExec.limit(),
+                        (List<Attribute>) lists[0],
+                        (List<Stat>) lists[1]
+                    );
+                }
+            }
+            return plan;
+        }
+
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        private List[] pushableStats(AggregateExec aggregate) {
+            AttributeMap<Stat> stats = new AttributeMap<>();
+            var lists = new List[] { new ArrayList<Attribute>(), new ArrayList<Stat>() };
```

Comment:
Instead of a record class, I opted for a vanilla array since the return is used only by the calling method.